src/requirements/Farcaster/hooks/useFarcasterAction.tsx

```diff
@@ -0,0 +1,59 @@
+import { Text } from "@chakra-ui/react"
+import useMembershipUpdate from "components/[guild]/JoinModal/hooks/useMembershipUpdate"
+import useUser from "components/[guild]/hooks/useUser"
+import useShowErrorToast from "hooks/useShowErrorToast"
+import useSubmit from "hooks/useSubmit"
+import useToast from "hooks/useToast"
+import { useFetcherWithSign } from "utils/fetcher"
+
+export default function useFarcasterAction(
+  roleId: number,
+  action: "follow" | "like" | "recast"
+) {
+  const { farcasterProfiles, id } = useUser()
+  const fetcherWithSign = useFetcherWithSign()
+  const { triggerMembershipUpdate } = useMembershipUpdate()
+  const toast = useToast()
+  const showErrorToast = useShowErrorToast()
+
+  return useSubmit(
+    async (param?: string) => {
+      if (!param) {
+        throw new Error("Unknown Farcaster action")
+      }
+      if (!farcasterProfiles?.[0]?.fid) {
+        throw new Error("First connect ypur Farcaster account")
```

Comment:
tpyo :kappa:

Reply:
Nice catch! Fixed in https://github.com/guildxyz/guild.xyz/pull/1322/commits/58cd9a648cadb7849db6b792616294d5e296d439